src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,79 @@
+import { useContext, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
-import { Loader } from './Loader';
+import { getPeople } from '../api';
 import { PeopleTable } from './PeopleTable';
+import { Loader } from './Loader';
+import { PeopleContext } from '../Context';
 
 export const PeoplePage = () => {
+  const {
+    dataComes,
+    loading,
+    err,
+    setloading,
+    setDataComes,
+    setPeople,
+    setErr,
+  } = useContext(PeopleContext);
+
+  useEffect(() => {
+    setloading(true);
+    getPeople()
+      .then((freshPeople) => {
+        setPeople(freshPeople);
+        if (freshPeople.length > 0) {
+          setDataComes('Exist');
+        } else {
+          setDataComes('Empty');
+        }
+      })
+      .catch((er) => {
+        setErr(er);
+        setDataComes('NO');
+      })
+      .finally(() => {
+        setloading(false);
+      });
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const { slug } = useParams();
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+          {(dataComes === 'Exist' && !loading) && (
+            <div className="column is-7-tablet is-narrow-desktop">
+              <PeopleFilters />
+            </div>
+          )}
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
-
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+              {loading && <Loader />}
 
-              <p data-cy="noPeopleMessage">
-                There are no people on the server
-              </p>
+              {err && (
+                <p data-cy="peopleLoadingError">
+                  Something went wrong
+                </p>
+              )}
 
-              <p>There are no people matching the current search criteria</p>
+              {dataComes === 'Empty' && (
```

Comment:
'Empty', 'Exist' ... and so other options put inside some enums
in order to understand what options can be